### DISCUSSION TITLE
Alternative Mouse Layer with Btn1 and Btn2 swapped.

### DISCUSSION THREAD

**Discussion by 10bn** [manna-harbour/miryoku#175] [Ideas]
As someone who has used a normal mouse almost his entire life I'd like to suggest an alternative mouse button order which is anlog to the default button order on a normal desktop mouse.

**Comment by manna-harbour** [comment]
Thanks for the suggestion!

I'll use a standard where differing from the standard wouldn't be a significant enough improvement, such as with the numpad arrangement. In this case, I think having the primary mouse button on the primary thumb is a significant improvement, but it is at the expense of differing from the standard. In general, alternative layout options are to support existing muscle (or even conceptual) memory, so this is a good candidate.

Actually, button order is one of the only areas of the layout that changed since the first version of Miryoku, and the only one that has flip-flopped. I switched to the standard order for a while, and it was nice having the compatibility, but the motivation for the change was actually to make the left button easier to reach with one hand on an OLED cover mounted trackpoint. But now with buttons on both hands of the Button layer, and the other hand being easier to use anyway, that's no longer a concern, and the primary thumb means no movement for the most common clicks. But an option should have been added when it was first changed.

I wasn't going to add any additional alternative layout options until after auto generating the alternatives and selection, as it's becoming awkward to maintain manually, but this can be implemented just by swapping macros in the same way as with the clipboard options, so it can be added now. 

I'll add an entry to https://github.com/manna-harbour/miryoku/issues/90. In the mean time you could make the change manually by substituting custom Mouse and Button layers using https://github.com/manna-harbour/miryoku/discussions/85.

**Comment by manna-harbour** [comment]
Here it is for Miryoku QMK: https://github.com/manna-harbour/miryoku_qmk/tree/miryoku-mousebuttons-tmp. Build with `MIRYOKU_MOUSEBUTTONS=STANDARD`. Here's a sample build run: https://github.com/manna-harbour/miryoku_qmk/actions/runs/3637152121.

It's in a temporary branch for now until it is added to the docs and the other implementations.

**Reply by 10bn** [reply]
It's great to see this being added. Thank you for picking this up.
